src/ocm/components/AddHosts/day2Wizard/Day2ClusterDetails.tsx

```diff
@@ -26,15 +50,47 @@ export const Day2ClusterDetails = () => {
   } = day2DiscoveryImageDialog;
   const wizardContext = useDay2WizardContext();
   const day1CpuArchitecture = cluster.cpuArchitecture as CpuArchitecture;
-
   const { isMultiCpuArchSupported } = useOpenshiftVersions();
+  const [initialValues, setInitialValues] = React.useState<Day2ClusterDetailValues>();
+  const [isSubmitting, setSubmitting] = React.useState(false);
+
+  React.useEffect(() => {
+    const doItAsync = async () => {
+      const initialValues = await getDay2ClusterDetailInitialValues(cluster.id);
+      setInitialValues(initialValues);
+    };
+    void doItAsync();
+  }, [cluster.id]);
 
   const handleSubmit = React.useCallback(
-    (values: Day2ClusterDetailValues) => {
-      wizardContext.setSelectedCpuArchitecture(values.cpuArchitecture);
-      wizardContext.moveNext();
+    async (values: Day2ClusterDetailValues) => {
+      try {
+        setSubmitting(true);
+        const { data: infraEnvs } = await InfraEnvsAPI.list(cluster.id);
+        if (infraEnvs.every((infraEnv) => infraEnv.staticNetworkConfig)) {
+          await InfraEnvsService.updateAll(cluster.id, {
```

Comment:
Also, let's please move the logic to Services, so instead of performing this update in the Render components, let's call a function in the corresponding Service which handles the logic appropriately.